kcoin/genesis/genesis_generator_test.go

```diff
@@ -0,0 +1,300 @@
+	if err != nil {
+		t.Fatalf("Error unmarshalling json genesis with error: %s", err.Error())
+	}
+
+	assertEqualGenesis(t, expectedGenesis, generatedGenesis)
+}
+
+//assertEqualGenesis checks if two genesis are the same, it ignores some fields as the timestamp that
+//will be always different when it is generated.
+func assertEqualGenesis(t *testing.T, expectedGenesis *core.Genesis, generatedGenesis *core.Genesis) {
+	assert.Equal(t, expectedGenesis.ExtraData, generatedGenesis.ExtraData)
+	assert.Equal(t, expectedGenesis.Config, generatedGenesis.Config)
+	assert.Equal(t, expectedGenesis.GasLimit, generatedGenesis.GasLimit)
+	assert.Equal(t, expectedGenesis.GasUsed, generatedGenesis.GasUsed)
+	assert.Equal(t, expectedGenesis.Coinbase, generatedGenesis.Coinbase)
+	assert.Equal(t, expectedGenesis.ParentHash, generatedGenesis.ParentHash)
+
+	assert.Len(t, expectedGenesis.Alloc, len(generatedGenesis.Alloc))
+
+	address := DefaultSmartContractsOwner
+	expectedAlloc := core.GenesisAccount{Balance: new(big.Int).Mul(common.Big1, big.NewInt(params.Ether))}
+	assert.Equal(t, generatedGenesis.Alloc[address], expectedAlloc)
+}
+
+func TestOptionalValues(t *testing.T) {
+	baseCommand := GenesisOptions{
+		Network:                       "test",
+		MaxNumValidators:              "5",
+		UnbondingPeriod:               "5",
+		WalletAddressGenesisValidator: "0xe2ac86cbae1bbbb47d157516d334e70859a1bee4",
+		PrefundedAccounts: []PrefundedAccount{
+			{
+				WalletAddress: "0xe2ac86cbae1bbbb47d157516d334e70859a1bee4",
+				Balance:       "15",
+			},
+		},
+	}
+
+	t.Run("Consensus engine value", func(t *testing.T) {
+		baseCommand.ConsensusEngine = "tendermint"
+
+		generatedGenesis, err := GenerateGenesis(baseCommand)
+		if err != nil {
+			t.Fatalf("Error: %s", err.Error())
+		}
+
+		assert.NotNil(t, generatedGenesis.Config.Tendermint)
+	})
+
+	t.Run("Smart contracts owner", func(t *testing.T) {
+		customSmartContractOwner := "0xe2ac86cbae1bbbb47d157516d334e70859a1aaaa"
+		baseCommand.SmartContractsOwner = customSmartContractOwner
+
+		generatedGenesis, err := GenerateGenesis(baseCommand)
+		if err != nil {
+			t.Fatalf("Error: %s", err.Error())
+		}
+
+		bigaddr, _ := new(big.Int).SetString(customSmartContractOwner, 0)
+		address := common.BigToAddress(bigaddr)
+		expectedAlloc := core.GenesisAccount{Balance: new(big.Int).Mul(common.Big1, big.NewInt(params.Ether))}
+
+		assert.Equal(t, generatedGenesis.Alloc[address], expectedAlloc)
+	})
+
+	t.Run("Extra data", func(t *testing.T) {
+		extraDataStr := "TheExtradata"
+		baseCommand.ExtraData = extraDataStr
+
+		generatedGenesis, err := GenerateGenesis(baseCommand)
+		if err != nil {
+			t.Fatalf("Error: %s", err.Error())
+		}
+
+		expectedExtradata := make([]byte, 32)
+		expectedExtradata = append([]byte(extraDataStr), expectedExtradata[len(extraDataStr):]...)
+
+		assert.Equal(t, expectedExtradata, generatedGenesis.ExtraData)
+	})
+}
+
+func TestItFailsWithAnInvalidNetwork(t *testing.T) {
```

Comment:
isn't this a dup test from the beginning of the file, or just the name `TestMapNetwork`?